src/js/stores/SourceStore.js

```diff
@@ -0,0 +1,53 @@
+// src/stores/ContactStore.js
+
+import Dispatcher from '../dispatch/Dispatcher';
+import { EventEmitter } from 'events';
+
+const CHANGE_EVENT = 'change'; 
+
+class SourceStoreClass extends EventEmitter {
+
+/**
+ *  initialize source array
+ */
+  constructor() {
+    super();
+    this.sources = [];  
+  }
+
+ /**
+  * Get the souces from the dispatcher
+  */
+
+  getSources() {
+    return this.sources;
+  }
+
+/**
+ * Reset the  source object to an empty array
+ */
+  setSources() {
+    this.sources = null;
+    return this.sources;
+  }
+
+/**
+ * @param {*} action GET_SOURCES
+ * listening to the dispather to get action look GET_SOURCES action types */
+
+  handleActions(action) {
+    if (action.actionType === 'GET_SOURCES') {
+      this.sources = action.sources;
+      this.emit(CHANGE_EVENT);
+    }
+  }
+
+}
+
+const SourceStore = new SourceStoreClass();
+
+/* SouceStore register a callback for the dispatcher
+*/
+Dispatcher.register(SourceStore.handleActions.bind(SourceStore));
+
+export default SourceStore;
```

Comment:
Your `ArticleStore` and `BaseStore` have lots of similarities which are repeated in both files. You can abstract that into a single file which you can name `baseStore`. Then both `ArticleStore` and `BaseStore` can extend the baseStore. See example of abstracting the baseStore here:
![screen shot 2017-06-29 at 20 48 04](https://user-images.githubusercontent.com/10073273/27707186-50a81720-5d0c-11e7-805f-520e0a50abd0.jpg)

And example of extending the baseStore here:
![screen shot 2017-06-29 at 20 49 24](https://user-images.githubusercontent.com/10073273/27707232-7a5ad562-5d0c-11e7-9978-1e78f41b133e.jpg)